modules/ROOT/generate-docs.sh

```diff
@@ -2,33 +2,82 @@
 
 set -exuo pipefail
 
-cd $(dirname $0)
+cd $(dirname "${BASH_SOURCE[0]}")
 
 GIT_ROOT=$(git rev-parse --show-toplevel)
+cd "${GIT_ROOT}"
 
-PATH_TO_SYNC_GATEWAY=${PATH_TO_SYNC_GATEWAY:-${GIT_ROOT}/../sync_gateway/}
+# Get docs tag name from current major.minor version
+yq -V || brew install yq
+tag=$(yq -r .version antora.yml)-docs
+
+# Get upstream sync_gateway repo
+[[ -d ./sync_gateway ]] || git clone --no-checkout https://github.com/couchbase/sync_gateway.git
```

Comment:
Why are we only cloning conditionally?
If the directory already exists, will the `sparse-checkout` commands below work as expected?

Reply:
Good point. This line is left over after copying / pasting from the docs-analytics repo, where the sparse checkout persists after the script is run. Here, we delete the sparse checkout at the end of the script, so we can probably assume the directory doesn't already exist.

Reply:
(The point is that cloning can take a while; the `assembly.sh` script in Analytics assumes that if the directory is already there, it's probably the cloned repo)

Reply:
However, I've found in the Analytics repo that persisting the cloned upstream repo can lead to problems: usually git ignores the cloned upstream repo (good), but _sometimes_ the cloned upstream repo gets randomly added to commits in the Analytics repo (bad). That was why I decided in this case to delete the cloned upstream repo after the script has finished with it, and put up with the delay caused by cloning each time.

Reply:
Done